programs/bpf_loader/src/lib.rs

```diff
@@ -244,9 +254,11 @@ impl Executor for BPFExecutor {
         )?;
         {
             let compute_meter = invoke_context.get_compute_meter();
-            let (mut vm, heap_region) = match create_vm(
+            let is_jit_enabled = invoke_context.is_feature_active(&just_in_time_compilation::id());
+            let mut vm = match create_vm(
```

Comment:
Side note, at this point maybe we don't have to create a vm, just call `vm::execute(..)`?

Reply:
There is still some shared code in the VM to setup the `memory_mapping` for the interpreter and the JIT. That is a reason to keep the VM constructor.